### PR TITLE
fix fop docbook.xsl path for macos, add syntax version to example pro…

### PR DESCRIPTION
**Pull request by guozheng**
…to files, update macos build instructions, update .gitignore to ignore qt cache and stash

@estan 

here is the PR related to #269

### CODE REVIEW

**Comment by estan** [comment]
Thanks a lot for your investigations and this pull request. Will review.

I don't have access to a macOS machine, so I need all the help I can get :)

File: BUILDING.md [hunk 1]
```diff
@@ -38,22 +38,28 @@ future.
 ## Mac OS X
 
 ### Install Build Tools
+First, you need to have Xcode and Xcode Command Line Tools installed. You can follow [this instruction](https://railsapps.github.io/xcode-command-line-tools.html) to install both.
```
Comment (guozheng) [inline BUILDING.md]: sure, will do

Comment (guozheng) [inline BUILDING.md]: did a bit search but could not find official doc, normally people just install xcode from app store.

File: BUILDING.md [hunk 2]
```diff
@@ -38,22 +38,28 @@ future.
...
 
-If you do not have Homebrew, install it first, see [here](http://brew.sh) for instructions.
+Then, you need to install Homebrew, the mighty package manager on macos, see [Homebrew site](http://brew.sh) for instructions.
```
Comment (guozheng) [inline BUILDING.md]: sure

File: BUILDING.md [hunk 3]
```diff
@@ -38,22 +38,28 @@ future.
...
 brew update
-brew install qt5 protobuf
+brew install qt5 protobuf fop docbook-xsl
 brew link --force qt5
 export PROTOBUF_PREFIX=$(brew --prefix protobuf)
 git clone git@github.com:estan/protoc-gen-doc.git
 cd protoc-gen-doc
 qmake
 make
+cd examples
```
Comment (guozheng) [inline BUILDING.md]: this means sense, I can add a separate section for `Building the Examples`

File: examples/Makefile [hunk 4]
```diff
@@ -1,4 +1,11 @@
-DOCBOOK_XSL?=/usr/share/xml/docbook/xsl-stylesheets-1.79.1/fo/docbook.xsl
+UNAME := $(shell uname)
```
Comment (guozheng) [inline examples/Makefile]: ok, then I can move this into the doc instead

File: examples/proto/Booking.proto [hunk 5]
```diff
@@ -6,6 +6,7 @@
  *
  * Author: Elvis Stansvik
  */
+syntax = "proto2";
```
Comment (guozheng) [inline examples/proto/Booking.proto]: I can start learning the code base and see if I can offer some help. This tool is very nice and I really enjoy using it.

**Comment by guozheng** [comment]
@estan updated the PR, please take another look, thanks!

**Comment by estan** [comment]
Please see my recent addition to the README.md, copied below. My sincerest apologies for this.

```
To all users of `protoc-gen-doc`: This project is now unmaintained and
seeking a new maintainer. The reason for this is two-fold:

1. I'm not using Protocol Buffers myself anymore, and haven't done so
   for years.

2. I'm too busy with work and other projects to keep developing
   the tool, deal with feature requests and PRs.

Too all those who have outstanding pull requests and/or issues filed:
I'm really sorry, and thank you so much for your contributions, but I
simply don't have time for the project anymore, even if it is quite
low-maintenance.

Are you a user of `protoc-gen-doc` who feel you could take over? Great!
Please get in touch with me privately and we can discuss how to best
proceed (GitHub forking and transferring of Travis CI / AppVeyor / Open
Build Service accounts). The code is quite simple and only moderate C++
skills and some familiarity with Qt is needed.

Again, to all users: I'm sorry for this inconvenience, but it would be
dishonest of me to continue claiming maintainership of this project
when I don't have the time anymore.
```

**Pull request closed without merging**